### PR TITLE
[WFLY-9464] Some XACML tests fail with security manager

### PR DESCRIPTION
**Pull request by yersan**
This issue cannot be merged directly. It depends on some changes in two libraries `jboss-jaxb-api_spec` and `security-xacml`

Apart from the two privileged blocks in the libraries, there are some additional permissions required at the application level.

Issue: https://issues.jboss.org/browse/WFLY-9464
JBEAP Issue: https://issues.jboss.org/browse/JBEAP-13576

It depends on: 
https://github.com/picketbox/security-xacml/pull/9
https://github.com/jboss/jboss-jaxb-api_spec/pull/9

### CODE REVIEW

**Comment by bstansberry** [comment]
@yersan Is this usable now?

**Comment by yersan** [comment]
Hi @bstansberry , not yet. We need two component upgrades in order to get this test working with secmgr enabled: jboss xacml (minimum 2.0.10) and Jaxb API (minimum 1.0.5)

These are the issues that address the component upgrades
Jaxb API upgrade: https://issues.jboss.org/browse/WFLY-6595 
Xacml upgrade: https://issues.jboss.org/browse/WFLY-9589

**Comment by kabir** [comment]
@yersan I am adding the 'hold' label. Let me know when it is ready (or if it already is!)

**Comment by stuartwdouglas** [comment]
@yersan it looks like this could be merged now, although the tests will not pass with the security manager until the components are upgraded?

**Comment by yersan** [comment]
Hi @stuartwdouglas , exactly, the tests under security manager will not pass until getting jboss xacml (minimum 2.0.10) and Jaxb API (minimum 1.0.5) in Wildfly. 

Honestly, I'm not sure if it should be merged now or we should wait until getting those versions already in the project, the changes in this PR were done for enabling the test under a sec manager and require the Jaxb/xacml version upgrades, maybe we should wait for those upgrades before merging this, or we can merge it and wait for the upgrades, after all the tests were failing under sec manager before.

**Comment by kabir** [comment]
We should wait, so the 'hold' label is still valid :)

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/security/xacml/EjbXACMLAuthorizationModuleTestCase.java [hunk 1]
```diff
@@ -160,7 +163,13 @@ private static JavaArchive createJar(final String archiveName, final String secu
                 .addAsResource(EjbXACMLAuthorizationModuleTestCase.class.getPackage(),
                         XACMLTestUtils.TESTOBJECTS_POLICIES + "/ejb-xacml-policy.xml", "xacml-policy.xml")
                 .addAsManifestResource(EjbXACMLAuthorizationModuleTestCase.class.getPackage(),
-                        XACMLTestUtils.TESTOBJECTS_CONFIG + "/jboss-ejb3.xml", "jboss-ejb3.xml");
+                        XACMLTestUtils.TESTOBJECTS_CONFIG + "/jboss-ejb3.xml", "jboss-ejb3.xml")
+                .addAsResource(createPermissionsXmlAsset(
+                        new RuntimePermission("accessDeclaredMembers"),
+                        new RuntimePermission("getClassLoader"),
+                        new ElytronPermission("getSecurityDomain"),
+                        new ReflectPermission("suppressAccessChecks")
```
Comment (hkalina) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/security/xacml/EjbXACMLAuthorizationModuleTestCase.java]: mentioned RuntimePermissions should not be required to be assigned to client code - I am going to add required privileged block into XACML as discussed in https://issues.jboss.org/browse/WFLY-10190

Comment (hkalina) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/security/xacml/EjbXACMLAuthorizationModuleTestCase.java]: Please remove the `RuntimePermission` from here - there will be added privileged block instead:
https://github.com/picketbox/security-xacml/pull/10
(discussion in https://issues.jboss.org/browse/WFLY-10190 )

Comment (yersan) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/security/xacml/EjbXACMLAuthorizationModuleTestCase.java]: @hkalina done, I've checked it with the latest XACML version including your picketbox/security-xacml#10 PR and those permissions are unnecessary. Thanks for pointing it out.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/security/xacml/JBossPDPInteroperabilityTestCase.java [hunk 2]
```diff
@@ -101,6 +107,16 @@ public static JavaArchive deployment() {
...
 
+        final String tempDir = System.getProperty("java.io.tmpdir");
+        final String testObjectsPolicies = AssetUtil.getClassLoaderResourceName(JBossPDPInteroperabilityTestCase.class.getPackage(), XACMLTestUtils.TESTOBJECTS_POLICIES);
+        jar.addAsResource(createPermissionsXmlAsset(
+                new FilePermission(testObjectsPolicies + "/-", "read"),
+                new FilePermission(tempDir + "/-", "read,write,delete"),
+                new PropertyPermission("java.io.tmpdir", "read"),
+                new ReflectPermission("suppressAccessChecks"),
+                new RuntimePermission("getClassLoader"),
+                new RuntimePermission("accessDeclaredMembers")
```
[the remark this answers is not in the document]
Comment (yersan) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/security/xacml/JBossPDPInteroperabilityTestCase.java]: @hkalina done, thanks

**Comment by yersan** [comment]
It still needs Xacml upgrade: https://issues.jboss.org/browse/WFLY-9589 in order to get the tests pass using secmgr. 

Since it is still on hold, I assume it is unnecessary to close it and open it again later once the upgrade is done.

**Comment by kabir** [comment]
Closing until we have all the required bits
